gem/lib/metasploit-payloads.rb

```diff
@@ -164,4 +182,30 @@ def self.warn_local_path(path)
       @local_paths << path
     end
   end
+
+  class << self
+    private
+    def list_meterpreter_things(&block)
+      things = [] # *things* is whatever is being enumerated (extension names, suffixes, etc.) as determined by the block
```

Comment:
Whilst I appreciate the fact that this is potentially designed to be a generic black box function in the idea of OOP style coding, naming it `things` is so vauge that based on the function name alone its really impossible to tell when this function should or should not be used.

As this function as far as I can tell is only used for finding things within a Metasploit directory and more specifically only deals with extensions names and suffixes, it may be better to name this `list_filtered_meterpreter_dir_results` or something similar.